identity_core/src/document.rs

```diff
@@ -56,28 +66,111 @@ impl DIDDocument {
-    pub fn add_key_pair(&mut self, key_pair: PublicKey) {
+    pub fn update_public_key(&mut self, key_pair: PublicKey) {
         self.public_key.push(key_pair);
+    }
 
-        self.update_time();
+    /// remove all of the public keys from the `DIDDocument`.
+    pub fn clear_public_keys(&mut self) {
+        self.public_key.clear();
     }
 
-    /// updates the `updated` fields time.
-    pub fn update_time(&mut self) {
-        self.updated = Utc::now().to_string();
+    /// sets in a new `auth` of type `Authentication` into the `DIDDocument`.
+    pub fn update_auth(&mut self, auth: Authentication) {
+        self.auth.push(auth);
+    }
+
+    /// remove all of the authentications from the `DIDDocument`.
+    pub fn clear_auth(&mut self) {
+        self.auth.clear();
+    }
+
+    /// sets in a new `assert` of type `Authentication` into the `DIDDocument`.
+    pub fn update_assert(&mut self, assert: Authentication) {
+        self.assert.push(assert);
+    }
+
+    /// remove all of the assertion methods from the `DIDDocument`.
+    pub fn clear_assert(&mut self) {
+        self.assert.clear();
+    }
+
+    /// sets in a new `verification` of type `Authentication` into the `DIDDocument`.
+    pub fn update_verification(&mut self, verification: Authentication) {
+        self.verification.push(verification);
+    }
+
+    /// remove all of the verification methods from the `DIDDocument`.
+    pub fn clear_verification(&mut self) {
+        self.verification.clear();
+    }
+
+    /// sets in a new `delegation` of type `Authentication` into the `DIDDocument`.
+    pub fn update_delegation(&mut self, delegation: Authentication) {
+        self.delegation.push(delegation);
     }
 
-    /// derive the did from the document.
+    /// remove all of the capability delegations from the `DIDDocument`.
+    pub fn clear_delegation(&mut self) {
+        self.delegation.clear();
+    }
+
+    /// sets in a new `invocation` of type `Authentication` into the `DIDDocument`.
+    pub fn update_invocation(&mut self, invocation: Authentication) {
+        self.invocation.push(invocation);
+    }
+
+    /// remove all of the capability invocations from the `DIDDocument`.
+    pub fn clear_invocation(&mut self) {
+        self.invocation.clear();
+    }
+
+    /// sets in a new `agreement` of type `Authentication` into the `DIDDocument`.
+    pub fn update_agreement(&mut self, agreement: Authentication) {
+        self.agreement.push(agreement);
+    }
+
+    /// remove all of the key agreements from the `DIDDocument`.
+    pub fn clear_agreement(&mut self) {
+        self.agreement.clear();
+    }
+
+    /// get the ID from the Document as a DID.
     pub fn derive_did(&self) -> crate::Result<DID> {
         self.id.to_did()
     }
+
+    /// Updates the `updated` time for the `DIDDocument`.
+    pub fn update_time(&mut self) {
+        self.updated = Some(Utc::now().to_rfc3339_opts(SecondsFormat::Millis, true));
```

Comment:
I've added a `Timestamp` type [here](https://github.com/iotaledger/identity.rs/pull/24/commits/fe047e0666fb43d54752242d69af12d27a55f0a2) to make this a little less repetitive and enforce the format.

That hasn't been merged yet so nothing to change for now

Reply:
I saw, yes, that's a great addition.